app/services/collection-indexes-service.js

```diff
@@ -139,3 +143,80 @@ exports.delete = function (id, callback) {
     });
 };
 
+/**
+ * Retrieves a collection index from the provided URL.
+ * This is expected to be a remote URL that does not require authentication.
+ */
+exports.retrieveByUrl = function(url, callback) {
+    if (!url) {
+        const error = new Error(errors.missingParameter);
+        return callback(error);
+    }
+
+    superagent.get(url).set('Accept', 'application/json').end((err, res) => {
+        if (err) {
+            if (err.response && err.response.notFound) {
+                const error = new Error(errors.notFound);
+                return callback(error);
+            } else if (err.response && err.response.badRequest) {
+                const error = new Error(errors.badRequest);
+                return callback(error);
+            } else if (err.code === 'ENOTFOUND') {
+                const error = new Error(errors.hostNotFound);
+                return callback(error);
+            } else if (err.code === 'ECONNREFUSED') {
+                const error = new Error(errors.connectionRefused);
+                return callback(error);
+            } else {
+                return callback(err)
+            }
+        }
+        else {
+            try {
+                // Parsing res.text handles both the content-type text/plain and application/json use cases
+                const collectionIndex = JSON.parse(res.text);
+                return callback(null, collectionIndex);
+            }
+            catch (err) {
+                return callback(err);
+            }
+        }
+    });
+}
```

Comment:
Note that this function is still required for the scheduler, even though the corresponding routes and controller functions should be deleted (since it doesn't need to be reachable from outside the REST API)